AppCenterData/AppCenterData/MSData.m

```diff
@@ -986,7 +986,7 @@ - (void)synchronizeLocalCacheWithCosmosDbWithToken:(MSTokenResult *)token
   id<MSRemoteOperationDelegate> strongDelegate;
   @synchronized(self) {
     strongDelegate = self.remoteOperationDelegate;
-    if (strongDelegate) {
+    if (strongDelegate && [strongDelegate respondsToSelector:@selector(data:didCompleteRemoteOperation:forDocumentMetadata:withError:)]) {
```

Comment:
This should also add a test to cover the new behavior

Reply:
added a test